test/test.js

```diff
@@ -1,131 +1,165 @@
+        res.body[0].should.have.a.property('href');
+        res.body[0].href.should.be.eql(config.THINGS_PATH + '/test-1');
+        done();
+      });
+  });
+
+  it('GET a property of a thing', (done) => {
+    chai.request(server)
+      .get(config.THINGS_PATH + '/test-1/properties/on')
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.have.a.property('on');
+        res.body.on.should.be.eql(false);
+        done();
+      });
+  });
+
+  it('fail to GET a non-existant property of a thing', (done) => {
+    chai.request(server)
+      .get(config.THINGS_PATH + '/test-1/properties/xyz')
+      .end((err, res) => {
+        res.should.have.status(500);
+        done();
+      });
+  });
+
+  it('fail to GET a property of a non-existent thing', (done) => {
+    chai.request(server)
+      .get(config.THINGS_PATH + '/test-1a/properties/on')
+      .end((err, res) => {
+        res.should.have.status(500);
+        done();
+      });
+  });
 
-var gateway = new Gateway(server_url);
-gateway.setDebug(opt.options.debug);
-
-gateway.getThings().then((things) => {
-  if (opt.options.thing) {
-    things = things.filter(thingMatches);
-  }
-  if (things.length === 0) {
-    console.log('No things found');
-    return;
-  }
-  if (opt.options.list) {
-    listAllThings(things);
-  }
-  if (opt.options.toggle) {
-    toggleAllThings(things);
-  }
-  if (opt.options.set) {
-    var set = opt.options.set.split('=');
-    if (set.length != 2) {
-      console.error('Expecting property=value to be passed to --set');
-    } else {
-      var propertyName = set[0];
-      var value = JSON.parse(set[1]);
-      setAll(things, propertyName, value);
-    }
-  }
-});
\ No newline at end of file
+  it('fail to set a property of a thing', (done) => {
+    chai.request(server)
+      .put(config.THINGS_PATH + '/test-1/properties/on')
+      .send({})
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
+
+  it('fail to set a property of a thing', (done) => {
+    chai.request(server)
+      .put(config.THINGS_PATH + '/test-1/properties/on')
+      .send({abc: true})
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
+
+  it('set a property of a thing', (done) => {
+    chai.request(server)
+      .put(config.THINGS_PATH + '/test-1/properties/on')
+      .send({on: true})
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.have.a.property('on');
+        res.body.on.should.be.eql(true);
+        done();
+      });
+  });
+
+  it('Verify property of a thing changed', (done) => {
+    chai.request(server)
+      .get(config.THINGS_PATH + '/test-1/properties/on')
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.have.a.property('on');
+        res.body.on.should.be.eql(true);
+        done();
+      });
+  });
+});
```

Comment:
I personally have always preferred hypens because you don't need to press the SHIFT key.